File: cdc-event-streaming/mastodon-consumer/src/main/java/com/onlyteo/sandbox/model/AccountEntity.java
package com.onlyteo.sandbox.model;

import jakarta.persistence.*;

import java.time.ZonedDateTime;

@Table(name = "ACCOUNT")
@Entity
public class AccountEntity {

    @Id
    @Column(name = "id", nullable = false, unique = true)
    private String id;
    @Column(name = "url", nullable = false)
    private String url;
    @Column(name = "avatar", nullable = false)
    private String avatar;
    @Column(name = "username", nullable = false)
    private String username;
    @Column(name = "account", nullable = false, unique = true)
    private String account;
    @Column(name = "display_name", nullable = false)
    private String displayName;
    @Column(name = "followers_count", nullable = false)
    private Integer followersCount;
    @Column(name = "following_count", nullable = false)
    private Integer followingCount;
    @Column(name = "statuses_count", nullable = false)
    private Integer statusesCount;
    @Column(name = "created_at", nullable = false)
    private ZonedDateTime createdAt;
    @OneToOne(mappedBy = "account")
    private StatusEntity status;

    public AccountEntity() {
    }

    public AccountEntity(String id,
                         String url,
                         String avatar,
                         String username,
                         String account,
                         String displayName,
                         Integer followersCount,
                         Integer followingCount,
                         Integer statusesCount,
                         ZonedDateTime createdAt) {
        this.id = id;
        this.url = url;
        this.avatar = avatar;
        this.username = username;
        this.account = account;
        this.displayName = displayName;
        this.followersCount = followersCount;
        this.followingCount = followingCount;
        this.statusesCount = statusesCount;
        this.createdAt = createdAt;
    }

    public String getId() {
        return id;
    }

    public String getUrl() {
        return url;
    }

    public String getAvatar() {
        return avatar;
    }

    public String getUsername() {
        return username;
    }

    public String getAccount() {
        return account;
    }

    public String getDisplayName() {
        return displayName;
    }

    public Integer getFollowersCount() {
        return followersCount;
    }

    public Integer getFollowingCount() {
        return followingCount;
    }

    public Integer getStatusesCount() {
        return statusesCount;
    }

    public ZonedDateTime getCreatedAt() {
        return createdAt;
    }

    public StatusEntity getStatus() {
        return status;
    }
}

File: spring-boot-kafka-cdc/mastodon-consumer/src/main/kotlin/com/onlyteo/sandbox/model/AccountEntity.kt
package com.onlyteo.sandbox.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import java.time.ZonedDateTime

@Table(name = "ACCOUNT")
@Entity
class AccountEntity(
    @Id @Column(name = "id", nullable = false, unique = true)
    var id: String? = null,
    @Column(name = "url", nullable = false)
    val url: String? = null,
    @Column(name = "avatar", nullable = false)
    var avatar: String? = null,
    @Column(name = "username", nullable = false)
    var username: String? = null,
    @Column(name = "account", nullable = false, unique = true)
    var account: String? = null,
    @Column(name = "display_name", nullable = false)
    var displayName: String? = null,
    @Column(name = "followers_count", nullable = false)
    var followersCount: Int? = null,
    @Column(name = "following_count", nullable = false)
    var followingCount: Int? = null,
    @Column(name = "statuses_count", nullable = false)
    var statusesCount: Int? = null,
    @Column(name = "created_at", nullable = false)
    var createdAt: ZonedDateTime? = null,
    @OneToOne(mappedBy = "account")
    val status: StatusEntity? = null
)
